packages/opal-common/opal_common/schemas/data.py

```diff
@@ -26,6 +27,15 @@ class DataSourceEntry(BaseModel):
         "PUT", description="Method used to write into OPA - PUT/PATCH"
     )
 
+class DataSourceEntryWithPollingInterval(DataSourceEntry):
+
+    # Periodic Update Interval
+    # If set, tells OPAL server how frequently to send message to clients that they need to refresh their data store from a data source
+    # Time in Seconds
+    periodic_update_interval: int = Field(
```

Comment:
Time in seconds is great - but a floating point number is a valid option , as opposed to limiting to int.
Also this field should be indicated as optional - i.e. `Optional[float]`

Reply:
So according to https://fastapi-utils.davidmontague.xyz/user-guide/repeated-tasks/#the-repeat_every-decorator its in "seconds" and I read that as implying whole second vales, so Int. I could be wrong and happy to change, just wanted to point out why its that value.

Reply:
The link you sent seems to indicate float as well:

> seconds: float : The number of seconds to wait between successive calls
https://fastapi-utils.davidmontague.xyz/user-guide/repeated-tasks/#keyword-arguments

Reply:
Speaking of which I think we should also pass the `wait_first=true`
As to not trigger immediately on load (duplicate call)

Reply:
> The link you sent seems to indicate float as well:
> 
> > seconds: float : The number of seconds to wait between successive calls
> > https://fastapi-utils.davidmontague.xyz/user-guide/repeated-tasks/#keyword-arguments

.... Reading is hard. Switching to Float

Reply:
> Speaking of which I think we should also pass the `wait_first=true` As to not trigger immediately on load (duplicate call)

I just came to the same conclusion. I've added that!